core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -527,7 +527,11 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties {
     // 6. Symbolic link
     val scenario6 = java.nio.file.Files.createSymbolicLink(new File(testDir, "scenario6")
```

Comment:
Hm .. should we better take a look at `Utils.createDirectory`, and fix this behavior to match with the previous one? I wonder how this difference affects the downstream that use `Utils.createDirectory` since this is used in core module.

I am fine with keeping this change if that's easier for you to investigate further, but we should probably revert this part if we find out that we should keep the original behaviour in `createDirectory` (which I believe is more likely).

Reply:
Let me get this in first .. but should probably follow up. I will leave this to you @dongjoon-hyun